node/src/indexer/tx_sender.rs

```diff
@@ -90,32 +143,53 @@ async fn submit_tx(
 async fn ensure_send_transaction(
     tx_signer: Arc<TransactionSigner>,
     indexer_state: Arc<IndexerState>,
-    method: String,
+    request: ChainSendTransactionRequest,
     params_ser: String,
     timeout: Duration,
     num_attempts: NonZeroUsize,
 ) {
     for _ in 0..num_attempts.into() {
-        let Some(nonce) = submit_tx(
+        if !submit_tx(
             tx_signer.clone(),
             indexer_state.clone(),
-            method.clone(),
+            request.method().to_string(),
             params_ser.clone(),
+            request.gas_required(),
         )
         .await
-        else {
-            // If the response fails to send immediately, wait a short period and try again
+        {
+            // If the transaction fails to send immediately, wait a short period and try again
+            metrics::MPC_OUTGOING_TRANSACTION_OUTCOMES
+                .with_label_values(&[request.method(), "local_error"])
+                .inc();
             time::sleep(Duration::from_secs(1)).await;
             continue;
         };
 
-        // If the transaction is sent, wait the full timeout then check if it got included
+        // Allow time for the transaction to be included
         time::sleep(timeout).await;
-        if indexer_state.has_nonce(nonce) {
-            metrics::MPC_NUM_SIGN_RESPONSES_INDEXED.inc();
-            return;
+        // Then try to check whether it had the intended effect
+        match confirm_tx_result(indexer_state.clone(), &request).await {
+            Ok(true) => {
+                metrics::MPC_OUTGOING_TRANSACTION_OUTCOMES
+                    .with_label_values(&[request.method(), "succeeded"])
+                    .inc();
+                return;
+            }
+            Ok(false) => {
+                metrics::MPC_OUTGOING_TRANSACTION_OUTCOMES
+                    .with_label_values(&[request.method(), "timed_out"])
+                    .inc();
+                continue;
+            }
+            Err(err) => {
+                metrics::MPC_OUTGOING_TRANSACTION_OUTCOMES
+                    .with_label_values(&[request.method(), "unknown"])
+                    .inc();
+                tracing::warn!(target:"mpc", %err, "encountered error trying to confirm result of transaction {:?}", request);
```

Comment:
Would this unconditionally be printed if the transaction is VotePk, VoteReshared, or Join?

How about changing `confirm_tx_result` to `should_tx_be_resubmitted` and return `Ok(false)` if it's VotePk, VoteReshared, or Join?

Reply:
Yeah that sounds good, it doesn't make sense to worry the node operator over this error. In the long run does it make sense to implement this logic for the other transaction types? Or would the coordinator loop resubmit them anyway.

Reply:
Went with an enum since `Ok(false)` makes "not implemented" indistinguishable from "don't resubmit, observed success". The metrics will be clearer this way.